doc/usr/source/9_other/10_inductive_validity_core.rst

```diff
@@ -0,0 +1,94 @@
+* ``--ivc_only_main_node <bool>`` (default ``false``\ ) -- Compute an IVC over the elements of the main node only
+* ``--ivc_all <bool>`` (default ``false``\ ) -- Compute all the Inductive Validity Cores
+* ``--ivc_approximate <bool>`` (default ``true``\ ) -- Compute an approximation (superset) of an IVC. Ignored if ``--ivc_all`` is ``true``
+* ``--ivc_smallest_first <bool>`` (default ``false``\ ) -- Compute the smallest IVC first. If ``--ivc_all`` is ``false``, compute the smallest IVC only
+* ``--ivc_must_set <bool>`` (default ``false``\ ) -- Compute the MUST set in addition to the IVCs
+* ``--print_ivc <bool>`` (default ``true``\ ) -- Print the inductive validity core computed
+* ``--print_ivc_complement <bool>`` (default ``false``\ ) -- Print the complement of the inductive validity core computed (= the elements that are not necessary to prove the system safe)
+* ``--minimize_program {no|valid_lustre|concise}`` (default ``no``\ ) -- Minimize the source Lustre program according to the inductive validity core(s) computed
+* ``--ivc_output_dir <string>`` (default ``<INPUT_FILENAME>``\ ) -- Output directory for the minimized programs
+* ``--ivc_uc_timeout <int>`` (default ``10``\ ) -- Set a timeout for each unsat core check sent to the solver
+* ``--ivc_precomputed_mcs <int>`` (default ``0``\ ) -- When computing all IVCs, determine the cardinality up to which MCS will be computed before starting to compute the IVCs
+
+Example
+-------
+
+Let's consider the following Lustre code:
+
+.. code-block:: none
+
+  contract fSpec(u,v: real) returns(r: real);
+  let
+      guarantee r >= 0.0;
+      guarantee true -> r >= pre(r);
+      guarantee r >= u;
+      guarantee r >= v;
+  tel;
+
+  node f(u, v : real) returns (r : real);
+  (*@contract import fSpec(u,v) returns (r) ; *)
+  var m1,m2: real;
+  let
+      m1 = if v > u then v else u;
+      m2 = if m1 > 0.0 then m1 else 0.0;
+      r = m2 -> if pre(r) > m1 then pre(r) else m1;
+  tel;
+
+  node main(x, y : real) returns (P : bool);
+  var a,b : real;
+  let
+      a = f(x,y);
+      b = f(y,x);
+      P = a >= x and a >= y and b >= x and b >= y;
+      --%PROPERTY P;
+  tel;
+
+If we are interesting in determining which guarantees of the contract ``fSpec`` of ``f`` are needed to prove ``P``,
+we should run this command:
+
+.. code-block:: none
+
+  kind2 <lustre_file> --ivc true --ivc_category contracts --ivc_only_main_node false --compositional true
+
+* ``--ivc_category contracts``: because we are only interested in minimizing the contract ``fSpec``
+* ``--ivc_only_main_node false``: because ``fSpec`` is not the contract of the main node, so we need to consider all nodes
+* ``--compositional true``: as we want to minimize the contract of ``f`` and not its implementation, we must enable the compositionnal reasonning
```

Comment:
must enable the compositionnal reasonning -> need to enable compositional analysis